ThirdPartyModules/Mirasvit/Credit.php

```diff
@@ -0,0 +1,190 @@
+     * @param EventsForThirdPartyModules   $eventsForThirdPartyModules
+     * @param State                        $appState
+     */
+    public function __construct(
+        Discount    $discountHelper,
+        State       $appState
+    ) {
+        $this->discountHelper = $discountHelper;
+        $this->appState       = $appState;
+    }
+
+    public function collectDiscounts($result,
+                                     $mirasvitStoreCreditHelper,
+                                     $mirasvitStoreCreditCalculationService,
+                                     $mirasvitStoreCreditCalculationConfig,
+                                     $mirasvitStoreCreditCalculationConfigLegacy,
+                                     $quote,
+                                     $paymentOnly)
+    {
+        list ($discounts, $totalAmount, $diff) = $result;
+        $this->mirasvitStoreCreditHelper = $mirasvitStoreCreditHelper;
+        $this->mirasvitStoreCreditCalculationService = $mirasvitStoreCreditCalculationService;
+        $this->mirasvitStoreCreditCalculationConfig = $mirasvitStoreCreditCalculationConfig;
+        $this->mirasvitStoreCreditCalculationConfigLegacy = $mirasvitStoreCreditCalculationConfigLegacy;
+
+        try {
+            $amount = abs($this->getMirasvitStoreCreditAmount($quote, $paymentOnly));
+            $currencyCode = $quote->getQuoteCurrencyCode();
+            $roundedAmount = CurrencyUtils::toMinor($amount, $currencyCode);
+
+            $discounts[] = [
+                'description'       => 'Store Credit',
+                'amount'            => $roundedAmount,
+                'discount_category' => Discount::BOLT_DISCOUNT_CATEGORY_STORE_CREDIT,
+                'discount_type'     => $this->discountHelper->getBoltDiscountType('by_fixed'), // For v1/discounts.code.apply and v2/cart.update
+                'type'              => $this->discountHelper->getBoltDiscountType('by_fixed'), // For v1/merchant/order
+            ];
+
+            $diff -= CurrencyUtils::toMinorWithoutRounding($amount, $currencyCode) - $roundedAmount;
+            $totalAmount -= $roundedAmount;
+        } catch (\Exception $e) {
+            $this->bugsnagHelper->notifyException($e);
+        } finally {        
+            return [$discounts, $totalAmount, $diff];
+        }
+    }
+    
+    /**
+     * @param      $quote
+     *
+     * @param bool $paymentOnly
+     *
+     * @return float
+     */
+    private function getMirasvitStoreCreditAmount($quote, $paymentOnly = false)
+    {
+        $miravitBalanceAmount = $this->getMirasvitStoreCreditUsedAmount($quote);
+            
+        if (!$paymentOnly) {
+            // For old version of Mirasvit Store Credit plugin,
+            // $miravitCalculationConfig could be empty,
+            // so we use the instance of \Mirasvit\Credit\Service\Config\CalculationConfig instead.
+            $miravitCalculationConfig = !empty($this->mirasvitStoreCreditCalculationConfig)
```

Comment:
I am not sure if this works. If $mirasvitStoreCreditCalculationConfig empty, we just don't run this code.
How about creating two different event listeners - one for the last version or one for the legacy?

You can introduce an additional parameter to run different methods from the one class or create a class for legacy support and child class of it for last version support, as you prefer.

Reply:
Well, now I group them into one line, `Mirasvit\Credit\Api\Config\CalculationConfigInterface|Mirasvit\Credit\Service\Config\CalculationConfig` , so we have a delimiter | between the two class names, and the code can check either of them exist or instance can be created, and in this way it can cover legency version in some contexts.